test/integration/model/upsert.test.js

```diff
@@ -487,6 +487,34 @@ describe(Support.getTestDialectTeaser('Model'), () => {
             });
         });
       }
+
+      if (dialect.match(/^postgres/)) {
+        it('works when deletedAt is Infinity and part of primary key', function() {
+          const User = this.sequelize.define('User', {
+            name: {
+              type: DataTypes.STRING,
+              primaryKey: true
+            },
+            address: DataTypes.STRING,
+            deletedAt: {
+              type: DataTypes.DATE,
+              primaryKey: true,
+              allowNull: false,
+              defaultValue: Infinity
+            }
+          }, {
+            paranoid: true
+          });
+
+          return User.sync({ force: true }).then(() => {
+            return User.build({ name: 'user1' })
+              .save()
+              .then(() => {
+                return User.upsert({ name: 'user1', address: 'address' }, {logging: console.log});
```

Comment:
Is the logging important for the test? Otherwise it just spams the mocha output

Reply:
Good catch, I'll remove.